apps/src/lib/client/tx.rs

```diff
@@ -1467,3 +1468,32 @@ pub async fn submit_tx(
 ) -> Result<TxResponse, error::Error> {
     tx::submit_tx(client, to_broadcast).await
 }
+
+pub async fn gen_ibc_shielded_transfer<C>(
+    client: &C,
+    ctx: &mut Context,
+    args: args::GenIbcShieldedTransafer,
+) -> Result<(), error::Error>
+where
+    C: namada::ledger::queries::Client + Sync,
+    C::Error: std::fmt::Display,
+{
+    if let Some(shielded_transfer) =
+        tx::gen_ibc_shielded_transfer(client, &mut ctx.shielded, args.clone())
+            .await?
+    {
+        let tx_id = shielded_transfer.masp_tx.txid().to_string();
+        let filename = format!("ibc_shielded_transfer_{}.memo", tx_id,);
+        let output_path = match &args.output_folder {
+            Some(path) => path.join(filename),
+            None => filename.into(),
+        };
+        let out = File::create(&output_path)
+            .expect("Should be able to create the out file.");
+        serde_json::to_writer_pretty(out, &Memo::from(shielded_transfer))
+            .expect("IBC memo should be deserializable.");
+    } else {
+        eprintln!("No shielded transfer for this IBC transfer.")
+    }
+    Ok(())
+}
```

Comment:
it should be possible to hold the intermediary memo in memory, and write it to the ibc transfer msg in a single CLI command, no? it's better from a UX perspective IMO

generating the shielded transfer memo as a separate command can still be useful though.

Reply:
This is requested on the destination Namada. The memo is used by the source Namada.
Though a single CLI could request another Namada chain, for a non-Namada source chain, the separate command could be useful, I think.